### ISSUE TITLE
Trouble with optional route segments when in a folder

### ISSUE THREAD

**Issue by corygibbons** [kiliman/remix-flat-routes#28]
Having some trouble getting the new [optional route segments](https://github.com/kiliman/remix-flat-routes#optional-route-segments) working when nested in a folder:

This gives the output expected:
```
routes/
  hello.(thing).tsx
  index.tsx
```
```
<Routes>
  <Route file="root.tsx">
    <Route path="hello/thing?" file="routes/hello.(thing).tsx" />
    <Route index file="routes/index.tsx" />
  </Route>
</Routes>
```

However if nested in a folder we lose that route entirely:
```
routes/
  _folder+/
    hello.(thing).tsx
  index.tsx
```

```
<Routes>
  <Route file="root.tsx">
    <Route index file="routes/index.tsx" />
  </Route>
</Routes>
```

Using remix `v1.9.0` + remix-flat-routes `v0.5.1`

**Comment by kiliman** [comment]
Ok... aside from the syntax support, I haven't really played with optional routes. Also the `+` suffix was recently added, so may have issues with the new optional routes as well.

I'll have to see what's going on.

**Comment by brandonpittman** [comment]
It seems to be generating the correct route, but I just get 404s.

**Comment by kiliman** [comment]
Fixed in v0.5.6

**Comment by corygibbons** [comment]
Working great, thanks!
